src/rhash/cdreader.c

```diff
@@ -508,10 +508,129 @@ static void* cdreader_open_cue_track(const char* path, uint32_t track)
   return cdrom;
 }
 
+static void* cdreader_open_gdi_track(const char* path, uint32_t track)
+{
+  void* file_handle;
+  char buffer[1024], mode[16];
+  char file[256];
+  char* bin_path = "";
+  char track_data[256];
+  int current_track = 0;
+  char* ptr, * ptr2, * end;
+
+  int ignored_track_count = 0;
+  int offset = 0;
+  int done = 0;
+  size_t num_read = 0;
+  struct cdrom_t* cdrom = NULL;
+
+  file_handle = rc_file_open(path);
+  if (!file_handle)
+    return NULL;
+
+  file[0] = '\0';
+  do
+  {
+    num_read = rc_file_read(file_handle, buffer, sizeof(buffer) - 1);
+    if (num_read == 0)
+      break;
+
+    buffer[num_read] = 0;
+    if (num_read == sizeof(buffer) - 1)
+      end = buffer + sizeof(buffer) * 3 / 4;
+    else
+      end = buffer + num_read;
+
+    for (ptr = buffer; ptr < end; ++ptr)
+    {
+      if (ignored_track_count == 0)
+      {
+        // remove track counter until reach a newline
+        while (*ptr != '\r')
+        {
+          ++ptr;
+        }
+        ignored_track_count = 1;
+      }
+
+      ptr2 = ptr;
+
+      // if it finds content
+      while ((*ptr2 >= 97 && *ptr2 <= 122) || (*ptr2 >= 65 && *ptr2 <= 90) ||
+        (*ptr2 >= 48 && *ptr2 <= 57) || (*ptr2 == 32) || (*ptr2 == '.' || (*ptr2 == ' ')))
+      {
+        ++ptr2;
+      }
+
+      if (ptr2 - ptr != 0)
+      {
+        memcpy(track_data, ptr, ptr2 - ptr);  // operation to get track data
+        track_data[ptr2 - ptr] = '\0';
```

Comment:
you can eliminate the copy (and the `track_data` buffer) by just setting `*ptr2 = '\0'`

Reply:
Interesting, but is it worth it? Because i'd need to create another variable ptr3 to store progress from buffer reading.

Reply:
Why would you need a third pointer? `ptr2` would maintain the progress, and you already use it in that way when you set `ptr = ptr2` on line 575
```
*ptr2++ = '\0'; /* null terminate and advance */

/* 1st try to get file if it have double quotes*/
sscanf(ptr, "%d %*s %*s %s \"%[^\"]\" %*s", &current_track, sector_size, file);

if (current_track == (int)track)
{
    ...

    if (strlen(file) == 0)
        sscanf(track_data, "%*d %*s %*s %*s %s %*s", file);

    ...
}

ptr = ptr2;
```

Reply:
I undestand now, thanks for explaining.